src/tools/pkcs11-tool.c

```diff
@@ -263,8 +263,8 @@ static const struct option options[] = {
 #ifndef _WIN32
 	{ "test-fork",		0, NULL,		OPT_TEST_FORK },
 #endif
-	{ "use-locking",	0, NULL,		OPT_USE_LOCKING },
 #if defined(_WIN32) || defined(HAVE_PTHREAD)
+	{ "use-locking",	0, NULL,		OPT_USE_LOCKING },
```

Comment:
I think this should stay as it is, but the check should come inside of the handling of this operation to provide better error than "unknown option".

Reply:
The issue is that OPT_USE_LOCKING is not defined when HAVE_PTHREAD is not defined.

Reply:
Right. It would need some more modifications. So lets keep it this way.